service/src/test/java/greencity/service/HabitAssignServiceImplTest.java

```diff
@@ -287,12 +289,36 @@ void unenrollHabitThrowWrongIdException() {
     }
 
     @Test
-    void deleteHabitAssign() {
+    void changeStatusHabitAssign() {
         HabitAssign habitAssign = ModelUtils.getHabitAssign();
-        when(habitAssignRepo.findByUserIdAndHabitId(1L, 1L)).thenReturn(Optional.ofNullable(habitAssign));
-        assert habitAssign != null;
-        habitAssignService.deleteHabitAssign(1L, 1L);
-        verify(habitAssignRepo).delete(habitAssign);
+        habitAssign.getHabit().setHabitAssigns(generateHabitAssignList());
+        when(habitAssignRepo.findListByUserIdAndHabitIdAndStatusIsInProgress(1L, 1L))
+            .thenReturn(habitAssign.getHabit().getHabitAssigns());
+        habitAssignService.cancelAllHabitAssign(1L, 1L);
+        Mockito.verify(userShoppingListItemRepo).deleteByShoppingListItemsByHabitAssignId(1L);
+        assert HabitAssignStatus.INPROGRESS != habitAssign.getHabit().getHabitAssigns().get(0).getStatus();
+        verify(habitAssignRepo, times(2)).save(any());
+    }
+
+    private List<HabitAssign> generateHabitAssignList() {
+        HabitAssign habitAssign1 = new HabitAssign();
+        habitAssign1.setId(1L);
+        habitAssign1.setStatus(HabitAssignStatus.INPROGRESS);
+
+        HabitAssign habitAssign2 = new HabitAssign();
+        habitAssign2.setId(2L);
+        habitAssign2.setStatus(HabitAssignStatus.ACQUIRED);
```

Comment:
I think this method should return `HabitAssignStatus.INPROGRESS` for all elements
Based on [this logic](https://github.com/ita-social-projects/GreenCity/pull/4865/files#diff-05169418d0f08eeb9efc32c9ae364ca45acf5015e13946bd7fffc632a1546bdcR209-R211) you will return habbits with only `INPROGRESS` status